### PR TITLE
Rewrite ARM builds to use arm-runner-action to resolve OrangePi5 images

### PR DESCRIPTION
**Pull request by judsonjames**
Directly related to https://github.com/PhotonVision/photonvision/issues/1142

This PR migrates all ARM builds from using the `generatePiImage.sh` script to using https://github.com/pguyot/arm-runner-action. The OrangePi image, when using the old script, would create a master boot record rather than a partitioned table when creating a loopback. Creating partitions manually was a hacky workaround that wouldn't scale to every ARM processor, but it was attempted.

Migrating to arm-runner-action seems to resolve the issue on my OrangePi5. I tested using the releases tab from my fork. I do not have an RPi or LL on-hand to test those platforms, but would appreciate others test it on their devices.

Builds for RPi and LL are available in the releases tab of my fork, linked below:
https://github.com/judsonjames/photonvision/releases/tag/Dev

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -293,11 +299,23 @@ jobs:
...
         with:
           name: jar-${{ matrix.artifact-name }}
-      # TODO- replace with the arm-runner action and run this inside of the chroot. but this works for now.
-      - name: Generate image
+      - uses: pguyot/arm-runner-action@v2
+        name: Generate image
+        id: generate_image
+        with:
+          base_image: ${{ matrix.image_url }}
+          image_additional_mb: 4096
```
Comment (mcm001) [inline .github/workflows/build.yml:307]: This number is hardcoded here, but it's also present in the matrix above

File: scripts/armrunner.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+DEST_PV_LOCATION=/opt/photonvision
+sudo mkdir -p $DEST_PV_LOCATION
+sudo cp $NEW_JAR ${DEST_PV_LOCATION}/photonvision.jar
+
+TARGET_WANTS=/etc/systemd/system/multi-user.target.wants
+sudo mkdir -p $TARGET_WANTS
+cd $TARGET_WANTS
+sudo bash -c "printf \
+\"[Unit]
```
Comment (mcm001) [inline scripts/armrunner.sh:16]: This service is missing the loud CPU flag that we comment out by default. Our generator has a known good version of this service file, so we might be able to delete this step entirely

Comment (judsonjames) [reply scripts/armrunner.sh]: That's what I thought too, but wasn't sure. I removed the service file write, and left a comment in the script saying that it assumes the image that arm-runner-action pulls has the servicefile pre-included.

**Review by amquake — APPROVED** [review]
Works on pi4

**Comment by trevnels** [comment]
I can confirm that the OPi image seems to be working as expected, and I'll be able to verify that the LL and Pi images work on Tuesday.

**Pull request merged**